### DISCUSSION TITLE
2 Applications in 1 Workspace

### DISCUSSION THREAD

**Discussion by anne-montoya** [aws/copilot-cli#5162] [General]
Currently we have a git repository with 2 branches that have their own copilot configuration folder. 

The 2 branches contain the same code so we want to merge the branches for easy maintenance.
But we still need to retain the 2 different copilot configuration.

I tried creating 2 applications on 1 branch but I got an error that says:
`workspace already registered with <application-name>`

Is this a valid use case? Can I do a workaround for this?

**Comment by huanjani** [comment]
Hi, @anne-montoya!

What do you think about one Copilot app with two different environments? Your workloads (services/jobs) could have different config per env. Would that work?

Another option is running `copilot app init` with a different name, so you could have your two different apps with the same code.

**Reply by anne-montoya** [reply]
Hello @huanjani, 

> What do you think about one Copilot app with two different environments?

As much as possible we don't want to change our current architecture/infra. 
For additional details, we already have 2 copilot app in different region containing almost 25 services each. These apps have 2 environment: test and production. Among those services, some share the same source code. So we want to just maintain one single source code (or one branch on git) but with different configuration for each region/app.

> Another option is running copilot app init with a different name

This is the use case I've tried on the example I gave above. I will put the whole error log.
```
✘ Workspace is already registered with application APP_1 instead of APP_2.
If you'd like to delete the application locally, you can delete the file at copilot/.workspace.
If you'd like to delete the application and all of its resources, run `copilot app delete`.
✘ workspace already registered with APP_1
```

**Comment by huanjani** [comment]
Ah, gotcha. Sorry I misunderstood earlier. (I think this is related to https://github.com/aws/copilot-cli/issues/3676--you could add your use case there if you'd like.)

We don't have a native solution (besides different envs/branches), but one hacky thing you could do is to manually go into your `copilot/.workspace` file, comment out the first app name, and add the second app name. When you run `copilot app init`, Copilot will be tricked into initializing the second app. This could be very error-prone and risky, so we don't really recommend it, but it is a workaround if you don't want to use envs to manage the two versions.
